driver-core/src/main/java/com/datastax/driver/core/QueryOptions.java

```diff
@@ -169,4 +170,31 @@ public QueryOptions setDefaultIdempotence(boolean defaultIdempotence) {
     public boolean getDefaultIdempotence() {
         return defaultIdempotence;
     }
+
+    /**
+     * Set whether the driver should re-prepare all cached prepared statements when
+     * it receives an onUp notification from the coordinator.
+     *
+     * If the node had a temporary network issue and had not completely crashed, then it
+     * is useless to re-prepare all queries on it.
+     *
+     * @jira_ticket JAVA-658
```

Comment:
This is not a recognized javadoc tag.
We use it in tests, but that's because the QA team has a custom tool to extract them. In production code, it will be ignored.